workflow/kfdrc_production_amplicon_architect.cwl

```diff
@@ -0,0 +1,286 @@
+
+  ### `tools/amplicon_architect.cwl`
+  This ties up all the prep work and outputs ecDNA candidates based on the seed regions provided
+  #### Inputs (critical):
+   - `data_repo`: Reference tar ball obtained from https://datasets.genepattern.org/?prefix=data/module_support_files/AmpliconArchitect/
+   - `data_ref_version`: enum, "GRCh38", "hg19", "GRCh37", "mm10", "GRCm38". Genome version in data repo to use. Should match `data_repo` downloaded.
+   - `bam`: Input bam/cram
+   - `downsample`: Recommended for anaylsis, recommended value be 10
+   - `output_basename`: output file name prefix
+   - `aa_bed`: Bed file from prepare-aa
+   - `coverage_stats`: Coverage stats file generated by prepare-aa
+   - `ref_cache`: _For cram input_, provide tar ball of output of running misc/seq_cache_populate.pl from samtools on the reference fasta
+   - `mosek_license_file`: This tool uses some software that requires a license file. You can get a personal or institutional one from https://www.mosek.com/license/request/
+   ### Outputs:
+   - `summary`: summary for all amplicons detected by AA
+   - `cycles`: text file for each amplicon listing the edges in the breakpoint graph, their categorization (sequence, discordant, concordant, source) and their copy counts
+   - `graph`: A text file for each amplicon listing the edges in the breakpoint graph, their categorization (sequence, discordant, concordant, source) and their copy counts
+   - `sv_pdf`: PDF image file displaying the SV view of AA
+   - `sv_png`: PNG image file displaying the SV view of AA
+
+  ### `tools/aa_classifier.cwl`
+  This tool classifies Amplicon Architect output to detect types of focal amplifications present
+  #### Inputs (critical):
+   - `data_repo`: Reference tar ball obtained from https://datasets.genepattern.org/?prefix=data/module_support_files/AmpliconArchitect/
+   - `data_ref_version`: enum, "GRCh38", "hg19", "GRCh37", "mm10", "GRCm38". Genome version in data repo to use. Should match `data_repo` downloaded.
+   - `cycles`: text file for each amplicon listing the edges in the breakpoint graph, their categorization (sequence, discordant, concordant, source) and their copy counts
+   - `graph`: A text file for each amplicon listing the edges in the breakpoint graph, their categorization (sequence, discordant, concordant, source) and their copy counts
+  #### Outputs:
+   - `amplicon_classification_profiles`: abstract classification of the amplicon
+   - `gene_list`: genes present on amplicons with each classification
+   - `ecDNA_cts`: not yet defined
+
+
+requirements:
+- class: MultipleInputFeatureRequirement
+- class: InlineJavascriptRequirement
+- class: ScatterFeatureRequirement
+
+inputs:
+  # Required
+  aa_data_repo: {type: File, doc: "Reference tar ball obtained from https://datasets.genepattern.org/?prefix=data/module_support_files/AmpliconArchitect/"}
+  aa_data_ref_version: {type: ['null', {type: enum, name: wgs_mode, symbols: ["GRCh38",
```

Comment:
why is it named wgs_mode?

Reply:
Ah, that's a holdover from copying from the CNVkit workflow